### PR TITLE
Add support for first-party cookie isolation

### LINKED ISSUE #1213: ghostery.com cookie handling in Firefox
Handle:
* storage partitioning
* first party isolation

### PR DESCRIPTION
**Pull request by smalluban**
Fixes #1213

### CODE REVIEW

File: extension-manifest-v3/src/utils/api.js [hunk 1]
```diff
@@ -38,6 +56,7 @@ export async function setCookie(name, value, durationInSec = COOKIE_DURATION) {
     name,
     value,
     expirationDate: Date.now() / 1000 + durationInSec,
+    firstPartyDomain: (await isFirstPartyIsolation()) ? DOMAIN : '',
```
Comment (chrmod) [inline extension-manifest-v3/src/utils/api.js:59]: are you sure that this param can be there on all platforms and with first party isolation disabled? I remember the API throwing exceptions when first party isolation was not supported, this is way MV2 adds this param conditionally firstPartyDomain

**Review by chrmod — CHANGES_REQUESTED** [review]
please confirm it was tested on all supported platforms

File: extension-manifest-v3/src/utils/api.js (hunk 1 above)
Comment (smalluban) [reply extension-manifest-v3/src/utils/api.js:59]: I can check again to be 100% sure, but from the MDN it looks like it should be fine to set `''` - this is the state when it is not supported.

Comment (smalluban) [reply extension-manifest-v3/src/utils/api.js]: In Safari is fine, but Chrome does not like the parameter, so you were right ;)

**Review by chrmod — APPROVED** [review]

**Pull request merged**